game_data_cleaner/main.py

```diff
@@ -29,22 +35,25 @@ def __init__(self) -> None:
             "publishers": [],
             "subcategories": [],
             "comments": [],
+            "expansions": [],
         }
 
-    def process_file_list(self):
+    def game_cleaning_chain(self):
+        self._process_file_list()
+        self._save_dfs_to_disk_or_s3()
+        
+
+    def _process_file_list(self):
         """Process the list of files in the S3 bucket
         This function will process the list of files in the S3 bucket
         and extract the necessary information from the XML files. The
         function will return None."""
 
         # list files in data dirty prefix in s3 using awswrangler
-        file_list = wr.s3.list_objects(
-            f"s3://{S3_SCRAPER_BUCKET}/{GAME_CONFIGS['output_xml_directory']}"
-        )
+        file_list = S3FileHandler().list_files(GAME_CONFIGS['output_xml_directory'])
         if not file_list:
-            file_list = os.listdir(
-                f"data/{GAME_CONFIGS['output_xml_directory']}"
-            )
+            local_files = os.listdir(f"data/{GAME_CONFIGS['output_xml_directory']}")
+            file_list = [x for x in local_files if '.gitkeep' not in x]
```

Comment:
It may be better to look for files we want using the extension, not remove those we don't.